replica/table.cc

```diff
@@ -1831,13 +1834,47 @@ table::table(schema_ptr schema, config config, lw_shared_ptr<const storage_optio
         tlogger.warn("Writes disabled, column family no durable.");
     }
     set_metrics();
+    recalculate_tablet_count_stats();
 }
 
 void table::update_effective_replication_map(locator::effective_replication_map_ptr erm) {
     auto old_erm = std::exchange(_erm, std::move(erm));
     if (old_erm) {
         old_erm->invalidate();
     }
+
+    recalculate_tablet_count_stats();
+}
+
+void table::recalculate_tablet_count_stats() {
+    if (!is_internal_keyspace(_schema->ks_name())) {
```

Comment:
Why this check?

Reply:
The outcome of the offline discussion with @amnonh is that the new metric is not exposed for tables from internal keyspace. In `table::set_metrics()` we expose it depending on `is_internal_keyspace`.

I thought, that there was no point in recalculating the value for these tables - currently we iterate over map and query shard_id of a given tablet. My idea was to save that time.

I can revert it if this is not needed.

Reply:
We still want the per-node metric to include all tablets, so I think filtering should not be done at this level.

Reply:
In the future we will move internal keyspaces to tablets too. Why do we exclude them in metrics @amnonh ?

Reply:
@tgrabiec It's a lot of metrics that are not very interesting; system tables are always local there are no tablets migration between nodes. We don't want to overwhelm the monitoring server with all those metrics.

This is especially important for users with only a few tables, in those situations, the user can choose to enable per-table metrics even if their monitoring server is not very large.

Reply:
They are interesting if you want to debug a problem caused by internal tables.

Reply:
@tgrabiec, and then you have the ```system.tablets```. This is why we removed the internal tables from the reporting. It does more harm than good. This is not just for tablets, this is for most per-table metrics.

Reply:
What's the problem with `system.tablets`?

Reply:
Would it help if we checked `uses_tablets()` instead of `is_internal_keyspace()` in `set_metrics()`? This way the metric would not be exported for tables that do not use tablets.

I have just tested it. The output seems to be the same in comparison with the previous version.

Is it future-proof? I mean, when the internal keyspace is moved to tablets, then the metric will include them.

Reply:
@amnonh @tgrabiec - do we have a decision here? Which version should be implemented? Do I understand correctly, that currently this is the last thing that needs to be adjusted in this PR?

Reply:
> Would it help if we checked `uses_tablets()` instead of `is_internal_keyspace()` in `set_metrics()`? This way the metric would not be exported for tables that do not use tablets.

Yes, it makes sense.

Reply:
> > Would it help if we checked `uses_tablets()` instead of `is_internal_keyspace()` in `set_metrics()`? This way the metric would not be exported for tables that do not use tablets.
> 
> Yes, it makes sense.

Okay - it is already implemented. I uploaded it with the last patch-set.